lian/src/lian_nav2/src/map.cpp

```diff
@@ -0,0 +1,96 @@
+#include"map.h"
+#include "rclcpp/rclcpp.hpp"
+#include "nav_msgs/msg/OccupancyGrid.hpp"
+
+Map::Map() : height(-1), width(-1), Grid(nullptr) {
+    this->declare_parameter("traversable_threshold", 50.0);
+    this->get_parameter("traversable_threshold", traversable_threshold);
+}
+Map::~Map()
+{	
+    if(Grid) {
+        for(int i = 0; i < height; i++) {
+            delete[] Grid[i];
+        }
+        delete[] Grid;
+    }
+}
+
+int * Map::operator [] (int i) {
+    return Grid[i];
+}
+
+const int * Map::operator [] (int i) const {
+    return Grid[i];
+}
+
+bool Map::CellIsTraversable(int curr_i, int curr_j) const {
+    return (Grid[curr_i][curr_j] != CN_OBSTL);
+}
+
+bool Map::CellIsObstacle(int curr_i, int curr_j) const {
+    return (Grid[curr_i][curr_j] == CN_OBSTL);
+}
+
+bool Map::CellOnGrid(int curr_i, int curr_j) const {
+    return (curr_i < height && curr_i >= 0 && curr_j < width && curr_j >= 0);
+}
+
+int Map::getHeight() const {
+    return height;
+}
+
+int Map::getWidth() const {
+    return width;
+}
+
+double Map::getCellSize() const {
+    return CellSize;
+}
+
+int cellThreshold(int cell_data[width]){
```

Comment:
The name of the parameter is looking like an array with index access, need to change to a normal name